prereise/gather/flexibilitydata/doe/doe_data.py

```diff
@@ -1,85 +1,108 @@
+    with zipfile.ZipFile(download_path+'/filter.zip', 'r') as fh:
+        fh.extractall(download_path)
 
     # delete and further extract
-    os.remove('data/filter.zip')
+    os.remove(download_path+'/filter.zip')
 
-    for i in os.listdir('data/'):
-        fh = tarfile.open('data/'+i)
-        fh.extractall('data')
+    for i in os.listdir(download_path):
+        fh = tarfile.open(os.path.join(download_path, i))
+        fh.extractall(download_path)
 
         fh.close()
-        os.remove('data/'+i)
+        os.remove(os.path.join(download_path, i))
 
     # cleanup
-    cleanup_directory('data/')
-
+    cleanup_directory(download_path)
 
-'''
-    aggregate_doe(root, out)
 
-Aggregate sector flexibilties by summing up the percentage flexibility from all sectors
-and store to output files
-'''
-def aggregate_doe(root, out):
-    if not os.path.exists(out):
-        os.mkdir(out)
+def aggregate_doe(root, out_path):
+    """Aggregate sector flexibilties by summing up the percentage flexibility from all sectors
+        and store to output csv file
+        
+    :param string root: the root directory containing raw de-compressed DOE flexibility data
+    :param string out_path: the output file where the aggregated data will be stored
+    :return: None
+    """
 
     all_folders = os.listdir(root)
+    
+
+    # initialize output container
+    eia_flex = pd.DataFrame(np.zeros((8808, len(all_folders))), columns=all_folders)
+
+
     for i in all_folders:
-        all_csvs = os.listdir(root+i)
-        # new container for total flexibility
-        file_flex = pd.read_csv(root + i + '/' + all_csvs[0], index_col=0)
-        total_flex = file_flex['Flexibility'].copy()
+        all_csvs = os.listdir(os.path.join(root, i))
         
+        # new column for total flexibility
+        file_flex = pd.read_csv(os.path.join(root, i, all_csvs[0]), index_col=0)
+
+        # assign index to the output df
+        if 'time' not in eia_flex.keys():
+            eia_flex['time'] = file_flex.index.copy()
+            eia_flex['time'] = pd.to_datetime(eia_flex['time'])
+            eia_flex.set_index('time', inplace=True)
+            
+        eia_flex[i] = eia_flex[i] + file_flex['Flexibility'].values
+        eia_flex.at[eia_flex.index==file_flex.index, i] = file_flex['Flexibility'].values
+
         for c in all_csvs[1:]:
-            fn = root + i + '/' + c
+            fn = os.path.join(root, i, c)
             file_flex = pd.read_csv(fn, index_col=0)
-            total_flex = total_flex + file_flex['Flexibility']
 
-        total_flex.to_csv(out_path + i + '.csv')
+            if file_flex.shape[0] == eia_flex.shape[0]:
+                eia_flex[i] = eia_flex[i] + file_flex['Flexibility'].values
+            else:                
+                eia_flex.loc[eia_flex.index.isin(pd.to_datetime(file_flex.index)), i] =  eia_flex.loc[eia_flex.index.isin(pd.to_datetime(file_flex.index)), i] + file_flex['Flexibility'].values
```

Comment:
I have not run black yet. There is only a handful of files that have this problem, so for the vast majority it should not be a concern.

Reply:
That doesn't seem fully reliable though since the file you choose to set the index might not be complete. If we are not expecting this index to change, could we set the index independent of the .csv files?

Reply:
We can potentially do. For this one specifically I checked the first file that it reads and it is complete.